modules/channel-web/src/backend/socket.ts

```diff
@@ -65,11 +71,76 @@ export default async (bp: typeof sdk, db: Database) => {
     next(undefined, false)
     // TODO Make official API (BotpressAPI.events.updateStatus(event.id, 'done'))
   }
+
+  const sendTyping = async (event, userId, conversationId) => {
+    const typing = parseTyping(event.payload.value)
+    const payload = bp.RealTimePayload.forVisitor(userId, 'webchat.typing', { timeInMs: typing, conversationId })
+    // Don't store "typing" in DB
+    bp.realtime.sendPayload(payload)
+    await Promise.delay(typing)
+  }
+
+  const applyChannelEffects = async (event: sdk.IO.OutgoingEvent, userId, conversationId) => {
+    let payload = event.payload
+
+    const {
+      __buttons,
+      __typing,
+      __trimText,
+      __markdown,
+      __dropdown,
+      __collectFeedback
+    } = payload.metadata as sdk.Content.Metadata
+
+    if (__typing) {
+      await sendTyping(event, userId, conversationId)
+    }
+
+    if (__trimText && !isNaN(__trimText)) {
+      payload.trimLength = __trimText
+    }
+
+    if (__markdown) {
+      payload.markdown = true
+    }
+
+    if (__collectFeedback) {
+      payload.collectFeedback = true
+    }
+
+    if (__buttons) {
+      payload = {
+        type: 'custom',
+        module: 'channel-web',
+        component: 'QuickReplies',
+        quick_replies: __buttons,
+        wrapped: {
+          type: event.type,
+          ..._.omit(event.payload, 'quick_replies')
+        }
+      }
+    }
+
+    if (__dropdown) {
+      payload = {
+        type: 'custom',
+        module: 'extensions',
```

Comment:
Since we use react-select for the feature, we have it in another module to keep the bundle size small. If we can implement it without altering the bundle size too much, then we can move it to channel web